gsPreCICE.h

```diff
@@ -305,16 +305,26 @@ struct mapCompare
         GISMO_ASSERT(coords.rows()==m_meshDims.at(meshName),"Dimension of the points ("<<coords.rows()<<") is not equal to the dimension of mesh "<<meshName<<"("<<m_meshDims.at(meshName)<<")\n");
         IDs.resize(coords.cols());
         const std::map<gsVector<T>,index_t,mapCompare> & map = m_maps.at(this->getMeshID(meshName));
+        T tolerance = 1e-6; // Define a tolerance for coordinate comparison
+
         for (index_t k=0; k!=coords.cols(); k++)
-#ifdef  NDEBUG
+    #ifdef  NDEBUG
             IDs.at(k) = map.at(coords.col(k));
-#else
+    #else
         {
-            typename std::map<gsVector<T>,index_t,mapCompare>::const_iterator it = map.find(coords.col(k));
-            GISMO_ASSERT(it!=map.end(),"Coordinate "<<coords.col(k).transpose()<<" is not registered in the vertex ID map of mesh "<<meshName<<".\nThis error could be because you registered points in the parametric domain for the mesh, but you try to read (i.e. evaluate) points defined in the physical domain or vice versa.");
-            IDs.at(k) = it->second;
+            bool found = false;
+            for (const auto& entry : map)
+            {
+                if ((entry.first - coords.col(k)).norm() < tolerance)
```

Comment:
Seems that the loop is somewhat like a `std::find_if` with a custom predicate. Can we use that? It's probably faster 

Also, check `gsMath` for `gsClose` to have comparisons depending on the numeric type (trying to make it independent of the tolerance)